src/py42/clients/settings/device_settings.py

```diff
@@ -414,42 +442,52 @@ def __str__(self):
 class TrackedFileSelectionList(UserList, object):
     """Helper class to track modifications to file selection lists."""
 
-    def __init__(self, manager, name, _list, changes_dict):
-        self.manager = manager
+    def __init__(self, backup_set, name, _list, changes_dict):
+        self.backup_set = backup_set
         self.name = name
         self.orig = list(_list)
         self.data = _list
         self._changes = changes_dict
 
+    def check_lock(self):
+        if self.backup_set.locked:
+            raise AttributeError("Unable to modify locked backup set.")
+
     def register_change(self):
-        self.manager._build_file_selection()
-        self.manager._build_regex_exclusions()
+        self.backup_set._build_file_selection()
+        self.backup_set._build_regex_exclusions()
         if set(self.orig) != set(self.data):
             self._changes[self.name] = show_change(self.orig, self.data)
         elif self.name in self._changes:
             del self._changes[self.name]
 
     def append(self, item):
+        self.check_lock()
```

Comment:
`lock_check` might make a good decorator

Reply:
I like it. done